scaladoc/src/dotty/tools/scaladoc/tasty/TypesSupport.scala

```diff
@@ -385,3 +399,21 @@ trait TypesSupport:
         case _ => false
 
     at.args.size == 2 && (!at.typeSymbol.name.forall(isIdentifierPart) || infixAnnot)
+
+  private def isPolyOrEreased(using Quotes)(tr: reflect.TypeRepr) =
+    Set("scala.PolyFunction", "scala.runtime.ErasedFunction")
+      .contains(tr.typeSymbol.fullName)
+
+  private def isContextualMethod(using Quotes)(mt: reflect.MethodType) =
+    mt.asInstanceOf[dotty.tools.dotc.core.Types.MethodType].isContextualMethod
```

Comment:
As a curiosity I will mention we've just added a MethodTypeKind to quotes api for that (but I think using the compiler directly here is better, and avoiding MethodTypeKind should allow us to use this scaladoc for LTS api).